solutions/instance/variables.tf

```diff
@@ -0,0 +1,90 @@
+variable "ibmcloud_api_key" {
+  type        = string
+  description = "The IBM Cloud API Token"
+  sensitive   = true
+}
+
+variable "existing_resource_group" {
+  type        = bool
+  description = "Whether to use an existing resource group."
+  default     = false
+}
+
+variable "resource_group_name" {
+  type        = string
+  description = "The name of a new or an existing resource group in which Cloud Object Storage instance will be provisioned."
+}
+
+variable "create_resource_key" {
+  description = "Set as true to create a new resource key for the Cloud Object Storage instance."
+  type        = bool
+  default     = false
+}
+
+variable "generate_hmac_credentials" {
+  description = "Set as true to generate an HMAC key in the resource key. Only used when create_resource_key is `true`."
+  type        = bool
+  default     = false
+}
+
+variable "resource_key_name" {
+  description = "The name of the resource key to be created."
+  type        = string
+  default     = "cos-resource-key"
+}
+
+variable "resource_key_role" {
+  description = "The role you want to be associated with your new resource key. Valid roles are 'Writer', 'Reader', 'Manager', 'Content Reader', 'Object Reader', 'Object Writer'."
+  type        = string
+  default     = "Manager"
+}
```

Comment:
When https://github.com/terraform-ibm-modules/terraform-ibm-cos/pull/522 is merged, I think we should use the new `resource_keys` variable instead of `create_resource_key`, `generate_hmac_credentials`, `resource_key_name` and `resource_key_role` variables.